File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        int n = intervals.size();
    vector<vector<int>> ans;
        for(int i=0; i< n;i++){
            if(ans.empty()){
                ans.push_back(intervals[i]);
            }
            else{
                vector<int> &v = ans.back();
                int endingValue = v[1];
                // [1,4] [2,6]
                //overlapping case
             if(intervals[i][0] <= endingValue){
                v[1] = max(endingValue, intervals[i][1]);
             }
             else{
                ans.push_back(intervals[i]);
             }

            }
        }
        return ans;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.js
var merge = function(intervals) {
    const result = [];

    if (intervals.length <= 1) return intervals;

    intervals.sort((a, b) => a[0] - b[0]);

    let interval = intervals[0];

    for (let i = 1; i < intervals.length; i++) {
        if (intervals[i][0] <= interval[1]) {
            interval[1] = Math.max(interval[1], intervals[i][1]);
        } else {
            result.push(interval);   // ✅ Push only when merge ends
            interval = intervals[i]; // \U0001f504 Start a new interval
        }
    }

    result.push(interval); // ✅ Push the last merged interval
    return result;
};
